File: app/schemas/fishing.py
from pydantic import BaseModel
from datetime import datetime


class FishRecordBase(BaseModel):
    fish_data: dict
    weights: list[dict]


class PlayerFishData(BaseModel):
    username: str
    uuid: str
    total_species: int
    total_catches: int
    last_updated: datetime
    fish_records: list[FishRecordBase]

File: frontend/src/app/services/fishing.ts
export interface FishData {
  name: string;
  rarity: string;
  climate: string;
  collection: string;
  catchTime: string;
  elusive: boolean;
}

export interface FishWeight {
  weight: string;
  firstCaught: string;
}

export interface FishRecordBase {
  fish_data: FishData;
  weights: FishWeight[];
  isExpanded?: boolean;
}

export interface PlayerFishData {
  username: string;
  uuid: string;
  total_species: number;
  total_catches: number;
  last_updated: string | Date;
  fish_records: FishRecordBase[];
}
